railgun/internal/cmd/rootcmd/servers.go

```diff
@@ -45,23 +45,32 @@ func StartAdmissionServer(ctx context.Context, c *manager.Config) error {
 	return nil
 }
 
-func StartProfilingServer(ctx context.Context, c *manager.Config) error {
+func StartDiagnosticsServer(ctx context.Context, port int, c *manager.Config) (diagnostics.Server, error) {
 	deprecatedLogger, err := util.MakeLogger(c.LogLevel, c.LogFormat)
 	if err != nil {
-		return err
+		return diagnostics.Server{}, err
 	}
 	logger := logrusr.NewLogger(deprecatedLogger)
 
-	if !c.EnableProfiling {
-		logger.Info("profiling server disabled")
-		return nil
+	if !c.EnableProfiling && !c.EnableConfigDumps {
+		logger.Info("diagnostics server disabled")
+		return diagnostics.Server{}, nil
```

Comment:
```suggestion
		return nil, nil
```

Reply:
Related to https://github.com/Kong/kubernetes-ingress-controller/pull/1589#discussion_r677646446 above. The return here is how we get the empty diagnostic server configuration (specifically the empty `ConfigDumps` field) that indicates we shouldn't collect diagnostic configs.